File: app/src/main/java/com/jqorz/planewar/listener/GameListener.java
package com.jqorz.planewar.listener;

/**
 * @author j1997
 * @since 2020/5/2
 */
public interface GameListener {

    void onHeroAttacked();

    void onHeroDie();

    void onEnemyDie(int type);

    void onEnemyAttacked(int type);

    void onGetBomb(int count);

    void onGameOver();
}

File: app/src/main/java/com/jqorz/planewar/listener/GameListener.kt
package com.jqorz.planewar.listener

/**
 * @author j1997
 * @since 2020/5/2
 */
interface GameListener {
    fun onHeroAttacked()
    fun onHeroDie()
    fun onEnemyDie(type: Int)
    fun onEnemyAttacked(type: Int)
    fun onGetBomb(count: Int)
    fun onGameOver()
}